intermediate-python-testing.md

```diff
@@ -195,34 +196,51 @@ def database(request):
     pg_user = DB_OPTS.get("username")
     pg_db = DB_OPTS["database"]
     init_postgresql_database(pg_user, pg_host, pg_port, pg_db)
-
-    @request.addfinalizer
-    def drop_database():
-        # Last arg is postgres version
-        drop_postgresql_database(pg_user, pg_host, pg_port, pg_db, 9.6)
 ```
 
 DataMaders: If you need to add extensions to your test database, see [the `database` fixture in `dedupe-service`](https://github.com/datamade/dedupe-service/blob/master/tests/conftest.py).
 
-The above example includes two new and crucial concepts. Notice the scope keyword argument in our `fixture` decorator. Scope determines how long the context your fixture creates, sticks around. In `pytest`, there are three options:
+### Scope, or How to avoid confusing dependencies between your tests
+
+Notice the scope keyword argument in the `fixture` decorator above. Scope determines how long the context your fixture creates, sticks around. In `pytest`, there are three options:
 
 * `session` – Fixture is run once for the entire test suite
 * `module` – Fixture is run once per test module
 * `function` – Fixture is run every time a test includes it
 
-Fixtures are function-scoped by default, but it would be silly to re-create a database for every test. Instead, use the `session` scope for fixtures that establish context that you aren’t going to change over the course of your testing, such as creating a database, initializing an application, or inserting immutable dummy data.
+**Fixtures are function-scoped by default.** You should use this default unless you have a compelling reason. For example, it would be silly to re-create a database for every test. It is appropriate to use the `session` scope for fixtures that establish context that you aren’t going to change over the course of your testing, such as creating a database, initializing an application, or inserting immutable dummy data.
+
+**Be cautious!** Changes made to broadly scoped fixtures persist for all other tests in that session or module. This can lead to confusing circumstances where you are unsure whether your test is failing or the fixture context you expect was changed by a previous test.
+
+To diagnose these unintended dependencies, run your tests in a random order with [`pytest-randomly`](https://pypi.python.org/pypi/pytest-randomly). Simply:
+
+```bash
+pip install pytest-randomly
+```
+
+Then run your tests as normal and behold! They will be shuffled, and you will be kept honest. (Toggle random functionality with the `--randomly-dont-reorganize` flag if you need a run in sequential order.)
```

Comment:
Wow! That's very simple. 

I'd move remove the parenthetical, since that statement seems essential. Also, add a clarifying statement: "This package runs the tests randomly by default. Toggle random..." 

Do you want to mention other arguments to use with `pytest-randomly`, like `--randomly-seed`?